src/structures/Interaction.js

```diff
@@ -270,9 +270,61 @@ class Interaction extends Base {
    * @returns {boolean}
    */
   isSelectMenu() {
+    return InteractionTypes[this.type] === InteractionTypes.MESSAGE_COMPONENT && typeof this.values !== 'undefined';
+  }
```

Comment:
This is a breaking change. It should return `true` only for string select menu interactions.

Reply:
no?

Reply:
As from my meaning selectMenu is every type of select

Reply:
But in currently uses only for string select menu and users doesn't expect to get other types.

Reply:
who doesnt expect it? lol
all of them are just select menus

Reply:
Before it checked whether the interaction type is `3` and the component type is `3`. But now it also checks for `5`, `6`, `7` and `8`. Users need to change `interaction.isSelectMenu()` to `interaction.isStringSelect()`, which they should do only in major versions, not minor or patch.

Reply:
Why was this marked as resolved when it's still breaking?

Reply:
Because I think current solution it better
When I asked some people they said that they think "select menus" are all types of slects

Reply:
What part of "it's still breaking" is unclear? This is a breaking change, it changes how the method behaves, it cannot be merged into v13.

Reply:
Resolved by: https://github.com/discordjs/discord.js/pull/8814/commits/134824a609e8fc0feb93c210a993acf0d4fb5e64